### ISSUE TITLE
Illegal attributes that begin with =

### ISSUE THREAD

**Issue by paj28** [jhy/jsoup#2143]
If we parse an attribute like `<test =foo=bar/>` Then in the DOM the attribute appears with the = sign, but when re-serialized it is generated without.

Code:
```
val doc = Jsoup.parse("<test =foo=\"bar\"/>")
for (elem in doc.select("test")) {
    for (attr in  elem.attributes()) {
        println(attr.key)
    }
}
println(doc.html())
```
Output:
```
=foo
<html>
 <head></head>
 <body>
  <test foo="bar" />
 </body>
</html>
```
This is problematic as if an application is doing validation on the key, to prevent XSS attacks, this can be a way to bypass the validation. I discovered this issue (in a lab environment, not a live app) just now.

The key for output can be accessed using `getValidKey()`. A potential solution to this is to normalise keys during parsing.

**Comment by jhy** [comment]
I'm not sure about this. Per the goal of parsing to the HTML spec / how current browsers do, parsing the key as "=foo" is correct, and so they should not be (further) normalized during the parse.

It might be better if instead of stripping invalid characters, we had a way to escape or otherwise rename the attribute name.

And jsoup's mechanism for XSS safety is in the Cleaner + Safelist combination. Anything other than a safelist (allowlist) approach is unsound. You could always re-parse the output to confirm the roundtrip, if required.

**Comment by jhy** [comment]
What about if in the serializer normalization phase, rather than stripping, we replace with `_`? That way the output would be like `<test _foo="bar" />` and therefore not a way to smuggle in the attribute if you're doing a block-list approach. And will perhaps be easier to understand when reviewing the output.

**Comment by paj28** [comment]
> jsoup's mechanism for XSS safety is in the Cleaner + Safelist combination

Fair enough, I guess that removes any pressing need to change the behaviour.

> rather than stripping, we replace with _

That sounds like a good solution. Avoids the risk and unlikely to break anything else.

**Comment by jhy** [comment]
Thanks, done
